public_html/noaa_cron.php

```diff
@@ -18,75 +18,112 @@
+// For testing, only process Boston to debug the API data
+$query = "SELECT code, lat, lon FROM location WHERE code = 'BOSTONMA'";
+// Regular operation - process all locations
+// $query = "SELECT code, lat, lon FROM location";
 
 $result = mysqli_query($link, $query);
 
 // It's important to have these outside the loops, so they all have the same time
 // this will cause problems for retrieving the data otherwise
-$today  = date('Y-m-d');     // format: 2012-01-09
-$now    = date('Y-m-d H:i:s');
-
-
-if($result) {
-    while($row = mysqli_fetch_array($result)) {
-        $code   = $row[0];
-        $lat    = $row[1];
-        $lon    = $row[2];
-
-        $data = get_highs_lows($today, '6', 'e', '24 hourly', $lat, $lon);
+$today = date('Y-m-d');     // format: 2012-01-09
+$now = date('Y-m-d H:i:s');
+
+// Initialize the REST client
+$noaaClient = new NoaaRestClient();
+
+$locations_processed = 0;
+
+/////////////////////////DEBUG/////////////////////////
+// Initialize array to collect debug data
+$debug_all_data = [];
+/////////////////////////DEBUG/////////////////////////
+
+while($row = mysqli_fetch_assoc($result)) {
+    $code = $row['code'];
+    $lat = $row['lat'];
+    $lon = $row['lon'];
+
+    echo "Processing $code ($lat, $lon)\n";
+    
+    // Use REST client instead of SOAP
+    $data = $noaaClient->get_highs_lows($today, '6', 'e', '24 hourly', $lat, $lon);
+    
+    if (!$data) {
+        error_log("Failed to get forecast data for location: $code ($lat, $lon)");
+        continue;
+    }
 
         $highs = $data['data']['parameters']['temperature'][0]['value'];
-        $lows  = $data['data']['parameters']['temperature'][1]['value'];
-        $text  = $data['data']['parameters']['weather']['weather-conditions'];
+    $lows = $data['data']['parameters']['temperature'][1]['value'];
+    $text = $data['data']['parameters']['weather']['weather-conditions'];
         $icons = $data['data']['parameters']['conditions-icon']['icon-link'];
         $dates = $data['data']['time-layout'][0]['start-valid-time'];
 
-        $range = range(0,5);
-        // error_log(var_export($data, 1));
-
-        foreach($range as $day) {
-            $fc_high = $highs[$day];
-            $fc_low  = $lows[$day];
-            $fc_text = $text[$day]['@attributes']['weather-summary'];
-            $fc_icon = $icons[$day];
-            $date    = $dates[$day];
-            $date    = substr($date, 0, 10);
-
-            $sql = sprintf(
-                "INSERT INTO noaa_weather
+    /////////////////////////DEBUG/////////////////////////
+    // DEBUG: Output the extracted data elements to see their structure
+    $debug_data = [
+        'location' => "$code ($lat, $lon)",
+        'highs' => $highs,
+        'lows' => $lows,
+        'text' => $text,
+        'icons' => $icons,
+        'dates' => $dates
+    ];
+    file_put_contents('/tmp/extracted_data.json', json_encode($debug_data, JSON_PRETTY_PRINT));
```

Comment:
Can we add a way to enable/disable this debug output? In production, we may not need/want to enable debugging, so it would be useful to be able to have it turned off by default, and only enable when needed.